src/providers/hover.ts

```diff
@@ -0,0 +1,100 @@
+import * as vscode from "vscode";
+import * as fs from "fs/promises";
+import * as path from "path";
+
+interface HoverData {
+  [keyword: string]: {
+    [block: string]: {
+      [description: string]: string; // dfn_name
```

Comment:
```suggestion
      [description: string]: string | string[]; // dfn_name
```

I'm not sure though. I just notice the first parts of the json file and I noticed that the `description` field points to `string[]`. I'd suggest you make it point to a single type, like `string[]` only. If you make it `string[]`, you no longer need the `Array.isArray` checks in your hover algorithm.